pallets/pallet-asset-switch/src/lib.rs

```diff
@@ -479,33 +481,7 @@ pub mod pallet {
 				Error::<T, I>::Liquidity
 			);
 
-			let asset_id_v4: AssetId = switch_pair.remote_asset_id.clone().try_into().map_err(|e| {
-				log::error!(
-					target: LOG_TARGET,
-					"Failed to convert asset ID {:?} into v4 `AssetId` with error {:?}",
-					switch_pair.remote_asset_id,
-					e
-				);
-				DispatchError::from(Error::<T, I>::Internal)
-			})?;
-			let remote_asset_fee_v4: Asset = switch_pair.remote_xcm_fee.clone().try_into().map_err(|e| {
-				log::error!(
-					target: LOG_TARGET,
-					"Failed to convert remote XCM asset fee {:?} into v4 `Asset` with error {:?}",
-					switch_pair.remote_xcm_fee,
-					e
-				);
-				DispatchError::from(Error::<T, I>::Xcm)
-			})?;
-			let destination_v4: Location = switch_pair.remote_reserve_location.clone().try_into().map_err(|e| {
-				log::error!(
-					target: LOG_TARGET,
-					"Failed to convert remote reserve location {:?} into v4 `Location` with error {:?}",
-					switch_pair.remote_reserve_location,
-					e
-				);
-				DispatchError::from(Error::<T, I>::Internal)
-			})?;
+			let switch_pair_v4 = SwitchPairInfoV4::<_, T, I>::try_from(switch_pair).map_err(DispatchError::from)?;
```

Comment:
I like these changes, but I'm thinking, that maybe the xcm version should be defined in the config? We are currently forcing version 4. Other teams might use a different version. Making the pallet generic against the xcm version requires more changes, which might be not worth it in regard to the effort.